doc/util.md

```diff
@@ -138,13 +148,78 @@ class MyBaseGame extends BaseGame {
  - `translateVector`: Just like `canvas translate` method, but takes a `Vector2` as an argument.
- - `renderAt` and `renderRotated`: if you are directly rendering to the `Canvas`, you can use these functions to easily manipulate coordinates to render things on the correct places. They change the `Canvas` transformation matrix but reset afterwards.
+ - `renderAt` and `renderRotated`: if you are directly rendering to the `Canvas`, you can use these
+  functions to easily manipulate coordinates to render things on the correct places. They change the
+  `Canvas` transformation matrix but reset afterwards.
+
+### Color
+
+Methods:
+ - `darken`: Darken the shade of the color by an amount between 0 to 1.
+ - `brighten`: Brighten the shade of the color by an amount between 0 to 1.
+
+### Image
+
+Methods:
+ - `pixelsInUint8`: Retrieves the pixel data as a `Uint8List`, in the `ImageByteFormat.rawRgba`
+ pixel format, for the image.
+ - `getBoundingRect`: Get the bounding rectangle of the `Image` as a `Rect`.
+ - `size`: The size of an `Image` as `Vector2`.
+ - `darken`: Darken each pixel of the `Image` by an amount between 0 to 1.
+ - `brighten`: Brighten each pixel of the `Image` by an amount between 0 to 1.
+
+### Offset
+
+Methods;
+ - `toVector2`; Creates an `Vector2` from the `Offset`.
+ - `toSize`: Creates a `Size` from the `Offset`.
+ - `toPoint`: Creates a `Point` from the `Offset`.
+ - `toRect`: Creates a `Rect` starting in (0,0) and its bottom right corner is the [Offset].
+
+### Rect
+
+Methods:
+ - `toOffset`: Creates an `Offset` from the `Rect`.
+ - `toVector2`: Creates a `Vector2` starting in (0,0) and goes to the size of the `Rect`.
+ - `containsPoint` Whether this `Rect` contains a `Vector2` point or not.
+ - `intersectsSegment`; Whether the segment formed by two `Vector2`s intersects this `Rect`.
+ - `intersectsLineSegment`: Whether the `LineSegmet` intersects the `Rect`.
+ - `toVertices`: Turns the four corners of the `Rect` into a list of `Vector2`,
+
+### Size
+
+Methods:
+ - `toVector2`; Creates an `Vector2` from the `Size`.
+ - `toOffset`: Creates a `Offset` from the `Size`.
+ - `toPoint`: Creates a `Point` from the `Size`.
+ - `toRect`: Creates a `Rect` starting in (0,0) with the size of `Size`.
+
+### Vector2
+
+Methods:
+ - `toOffset`: Creates a `Offset` from the `Vector2`.
+ - `toPoint`: Creates a `Point` from the `Vector2`.
+ - `toRect`: Creates a `Rect` starting in (0,0) with the size of `Vector2`.
+ - `toPositionedRect`: Creates a `Rect` starting from [x, y] in the `Vector2` and has the size of
+  the `Vector2` argument.
+ - `lerp`: Linearly interpolates the `Vector2` towards another Vector2.
+ - `rotate`: Rotates the `Vector2` with an angle specified in radians, it rotates around the
+  optionally defined `Vector2`, otherwise around the center.
+ - `scaleTo`: Changes the length of the `Vector2` to the length provided, without changing
+  direction.
+ - `fromInts`: Create a `Vector2` with ints as input
```

Comment:
this is from the factory right? would probably divide outside of Methods, maybe Mehtods/Factories/Operators ?